src/lucky_router/path_normalizer.cr

```diff
@@ -0,0 +1,11 @@
+class LuckyRouter::PathNormalizer
+  DEFAULT_PATH_VARIABLE_NAME = ":path_variable"
+
+  def self.normalize(path_parts : Array(PathPart)) : String
+    path_parts.map { |path_part| normalize(path_part) }.join('/')
+  end
+
+  private def self.normalize(path_part : PathPart) : String
+    path_part.path_variable? ? DEFAULT_PATH_VARIABLE_NAME : path_part.name
```

Comment:
so that you don't have to look into the method, optional path variables (`?:foo`) and globs (named and unnamed) are path variables.